### ISSUE TITLE
${ocaml_bin} not usable since beta19

### ISSUE THREAD

**Issue by dra27** [ocaml/dune#681]
Commit 138c920069bc559c36f4e4f7803c457de8c5bf0b causes the following `jbuild`:

```
(jbuild_version 1)

(rule
  (with-stderr-to foo (echo ${ocaml_bin})))
```

and the command `jbuilder build foo` to fail with:

```
Error: exception Sys_error("Is a directory")
Backtrace:
Raised at file "src/dep_path.ml" (inlined), line 46, characters 24-55
Called from file "src/build_system.ml", line 91, characters 6-48
Called from file "src/fiber/fiber.ml", line 303, characters 6-18
```

That patch-set is quite large, so I'm hoping @diml will be able to identify the error faster than I will 🙂

**Comment by rgrinberg** [comment]
The issue here seems twofold to me:

* A dependency on the path specified by `${ocaml_bin}` is being added
* The path of `${ocaml_bin}` is a directory rather than a file. Hence it cannot be a dependency in the usual sense.

I think you should be able to fix your issue by doing something like `${path-no-dep:ocaml_bin}`. But I'm not sure how to proceed further here. From a brief glance, I see a few options:

* We skip depending on special paths which we know point to directories.
* We setup some glob rule to depend on all the entire contents of a directory.
* It's also tempting to simply never consider a PATH that is an argument to `echo` as a dependency. I'll see how feasible this is.

cc'ing @Chris00 here since he touched some of this code as well.

**Comment by diml** [comment]
> We skip depending on special paths which we know point to directories.

That seems fine.

**Comment by rgrinberg** [comment]
@diml what about the issue of `(echo ${file})` introducing a dependency on `${file}`. This isn't really correct.

**Comment by rgrinberg** [comment]
@diml fixed this in 1.0.0 I believe
